sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -148,7 +149,15 @@ object AggUtils {
       case ne: NamedExpression => ne
       case other => Alias(other, other.toString)()
     }
-    val distinctAttributes = namedDistinctExpressions.map(_.toAttribute)
+    // Ideally this should be done in `NormalizeFloatingNumbers`, but we do it here because
```

Comment:
could the normalized distinct expressions be combined with the `namedDistinctExpressions` above (so there's a single `match` block)?

Reply:
+1 for @abellina 's question.

Reply:
Looks okay to combine.